### PR TITLE
Upgrade to `ESLint v9`

### PR DESCRIPTION
**Pull request by risantos**
Upgrades the plugin to `ESLint` 9.

Additionally:
- Add `release-it`
- Update tests
- Simplify scripts

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    strategy:
+      matrix:
+        node: ['20', '22']
+
+    container:
+      image: node:${{ matrix.node }}-alpine
+
+    steps:
+    - uses: actions/checkout@v4
+    - run: yarn install --frozen-lockfile
```
Comment (satazor) [inline .github/workflows/tests.yaml]: Switch to npm instead of yarn

File: .travis.yml [hunk 2]
```diff
@@ -1,7 +0,0 @@
-language: node_js
```
Comment (satazor) [inline .travis.yml:1]: Adjust badge in readme. Search for Travis, you will find it.

File: package.json [hunk 3]
```diff
@@ -25,15 +25,24 @@
     "sql-parse": "^0.1.5"
   },
   "engines": {
-    "node": ">=4"
+    "node": ">=20"
   },
   "scripts": {
```
Comment (satazor) [inline package.json:37]: Add publish config public.
See esllint-config-uphold repo for an example

File: README.md [hunk 4]
```diff
@@ -1,7 +1,5 @@
 # eslint-plugin-sql-template
 
-[![Build Status](https://travis-ci.org/uphold/eslint-plugin-sql-template.svg?branch=master)](https://travis-ci.org/uphold/eslint-plugin-sql-template)
```
Comment (satazor) [inline README.md:3]: Use github badge

Comment (risantos) [reply README.md:3]: Thanks, added

**Review by satazor — APPROVED** [review]

**Review by fixe — APPROVED** [review]

**Pull request merged**